pkg/neg/syncers/utils.go

```
@@ -466,3 +467,63 @@ func makeEndpointBatch(endpoints negtypes.NetworkEndpointSet, negType negtypes.N
 	}
 	return endpointBatch, nil
 }
+
+// getPodLabelMap will return a json string of a label map extracted from a pod according to PodLabelPropagationConfig
+// The return map json has the pod label key as key and label value as value
+func getPodLabelMap(namespace, name string, podLister cache.Indexer, lpConfig negtypes.PodLabelPropagationConfig, recorder record.EventRecorder, logger klog.Logger) map[string]string {
```

This func eats the error from the Lister. It's better to pass the error up to the stack and let the caller decide what to do with the error.

---

Done. Add error returns for pod related issues.